hbase-server/src/main/java/org/apache/hadoop/hbase/master/normalizer/MergeNormalizationPlan.java

```diff
@@ -78,4 +82,30 @@ public void execute(Admin admin) {
       LOG.error("Error during region merge: ", ex);
     }
   }
+
+  @Override
+  public boolean equals(Object o) {
+    if (this == o) {
+      return true;
+    }
+
+    if (o == null || getClass() != o.getClass()) {
+      return false;
+    }
+
+    MergeNormalizationPlan that = (MergeNormalizationPlan) o;
+
+    return new EqualsBuilder()
+      .append(firstRegion, that.firstRegion)
+      .append(secondRegion, that.secondRegion)
+      .isEquals();
+  }
+
+  @Override
+  public int hashCode() {
```

Comment:
By introducing `hashCode()` and `equals()`, comparison is no more dependant on getter methods, can we also remove `getFirstRegion()` and `getSecondRegion()`? Although they are useful, but I don't think they will be used anymore. Thought?

Reply:
Plus, removal of getters further reduces the chances of mutation of `firstRegion` and `secondRegion`.

Reply:
Yeah, seems they're unused. As is `SplitNormalizationPlan#getRegionInfo()`. Seems a little strange to have a POJO without public accessors, to include these members in `toString` but not have accessors, but yeah, less data visibility is better data visibility.

Reply:
Actually, I think i'd like to keep them. I'm finding they're handy for some of the experimental changes I've been trying. I'd prefer keep them for now.